### PR TITLE
87 create field constants file

### PR DESCRIPTION
**Pull request by ImAPenguin56**

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -36,7 +36,7 @@ deploy {
 def deployArtifact = deploy.targets.roborio.artifacts.frcCpp
 
 // Set this to true to enable desktop support.
-def includeDesktopSupport = false
+def includeDesktopSupport = true
```
Comment (nickfrisch) [inline build.gradle:39]: Can run the command Change Desktop Build Confguration or similar name to set to false, or manually edit

File: src/main/cpp/Robot.cpp [hunk 2]
```diff
@@ -43,7 +45,7 @@ void Robot::RobotInit()
 
     ChassisConfigMgr::GetInstance()->InitChassis(static_cast<RobotConfigMgr::RobotIdentifier>(teamNumber));
     auto chassisConfig = ChassisConfigMgr::GetInstance()->GetCurrentConfig();
-
+    m_fieldConsts = new FieldConsts();
```
Comment (nickfrisch) [inline src/main/cpp/Robot.cpp:48]: Remove this testing code

File: src/main/cpp/Robot.cpp [hunk 3]
```diff
@@ -171,7 +173,17 @@ void Robot::TeleopInit()
         m_field->ResetField(); // ToDo:  Move to DriveTeamFeedback
     }
     **/
-
+    m_field->AddPose("Blue Speaker", m_fieldConsts->GetFieldElement(FieldConsts::FIELD_ELEMENT::BLUE_SPEAKER).ToPose2d());
```
Comment (nickfrisch) [inline src/main/cpp/Robot.cpp:176]: Remove from here to line 185, testing code that shouldn't be in main

File: src/main/cpp/Robot.h [hunk 4]
```diff
@@ -32,6 +32,7 @@ class AutonPreviewer;
 class RobotState;
 class SomeMech;
 class RobotDefinition;
+class FieldConsts;
```
Comment (nickfrisch) [inline src/main/cpp/Robot.h:35]: Can remove for now, testing code

File: src/main/cpp/Robot.h [hunk 5]
```diff
@@ -64,6 +65,7 @@ class Robot : public frc::TimedRobot
     AdjustableItemMgr *m_tuner;
     FMSData *m_fmsData;
     DragonField *m_field;
+    FieldConsts *m_fieldConsts;
```
Comment (nickfrisch) [inline src/main/cpp/Robot.h:68]: Can remove, testing code

File: src/main/cpp/utils/FieldConsts.cpp [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM,
+// DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE
+// OR OTHER DEALINGS IN THE SOFTWARE.
+//====================================================================================================================================================
+#include "FieldConsts.h"
+using namespace std;
+
+FieldConsts::FieldConsts()
```
Comment (nickfrisch) [inline src/main/cpp/utils/FieldConsts.cpp:18]: Rename to FieldConstants

File: src/main/cpp/utils/FieldConsts.h [hunk 7]
```diff
@@ -0,0 +1,58 @@
...
+//====================================================================================================================================================
+#pragma once
+#include <frc/geometry/Pose3d.h>
+#include <frc/geometry/Rotation3d.h>
+#include "units/angle.h"
+#include "units/base.h"
+class FieldConsts
+{
+public:
+    FieldConsts();
```
Comment (nickfrisch) [inline src/main/cpp/utils/FieldConsts.h:23]: Make FieldConstants a singleton

**Review by nickfrisch — CHANGES_REQUESTED** [review]
Make sure to update from main

File: src/main/cpp/Robot.cpp [hunk 8]
```diff
@@ -27,6 +27,8 @@
 #include <utils/logging/LoggerData.h>
 #include <utils/logging/LoggerEnums.h>
 #include <AdjustableItemMgr.h>
+#include <frc/geometry/Pose3d.h>
```
Comment (nickfrisch) [inline src/main/cpp/Robot.cpp:30]: This include can be removed

File: src/main/cpp/Robot.cpp [hunk 9]
```diff
@@ -27,6 +27,8 @@
...
+#include <utils\FieldConstants.h>
```
Comment (nickfrisch) [inline src/main/cpp/Robot.cpp:31]: This include can be removed

**Review by nickfrisch — CHANGES_REQUESTED** [review]

**Review by nickfrisch — CHANGES_REQUESTED** [review]
Update from main and validate build, approved after that

**Comment by reslip1** [comment]
> Update from main and validate build, approved after that

I merged changes from main that preventing merging and build is good.

**Review by reslip1 — APPROVED** [review]

**Pull request merged**